src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -240,6 +250,40 @@ private class BoolArrayEqualityComparer : IEqualityComparer<bool[]>
             }
         }
 
+        private class BlobTupleEqualityComparer : IEqualityComparer<Tuple<string, byte[], int>>
```

Comment:
Do we actually need to compare anything but the name? A readonly blob with same name but different content sounds like an internal compiler error. We would end up with the same symbol for two different blobs.